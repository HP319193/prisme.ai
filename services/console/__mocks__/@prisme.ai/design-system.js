export function ListItem({ children = null }) {
  return children;
}

export function SearchInput({ children = null }) {
  return children;
}

export function Input({ children = null }) {
  return children;
}

export function Space({ children = null }) {
  return children;
}

export function Title({ children = null }) {
  return children;
}

export function Button({ children = null }) {
  return children;
}

export function SidePanel({ children = null }) {
  return children;
}

export function Menu({ children = null }) {
  return children;
}

export function Dropdown({ children = null }) {
  return children;
}

export function Divider({ children = null }) {
  return children;
}

export function Avatar({ children = null }) {
  return children;
}

export function Collapse({ children = null }) {
  return children;
}

export function Feed({ children = null }) {
  return children;
}

export function Layout({ children = null }) {
  return children;
}

export function Section({ children = null }) {
  return children;
}

export function CollapseItem({ children = null }) {
  return children;
}

export function PageHeader({ children = null }) {
  return children;
}

export function Row({ children = null }) {
  return children;
}

export function Col({ children = null }) {
  return children;
}

export function MenuTab({ children = null }) {
  return children;
}

export function Table({ children = null }) {
  return children;
}

export function Text({ children = null }) {
  return children;
}

export function TagEditable({ children = null }) {
  return children;
}

export function EditableTitle({ children = null }) {
  return children;
}

export function Loading({ children = null }) {
  return children;
}

export function Modal({ children = null }) {
  return children;
}

export function Card({ children = null }) {
  return children;
}

export function Switch({ children = null }) {
  return children;
}

export function Tooltip({ children = null }) {
  return children;
}

export const notification = {
  error: jest.fn(),
  confirm: jest.fn(),
  info: jest.fn(),
  success: jest.fn(),
};

export function Select({ children = null }) {
  return children;
}

export function Popover({ children = null }) {
  return children;
}

export function LocalizedInput({ children = null }) {
  return children;
}

export function SchemaForm({ children = null }) {
  return children;
}

export function SchemaFormDescription({ children = null }) {
  return children;
}

export const schemaTypes = [
  'string',
  'localized:string',
  'number',
  'localized:number',
  'boolean',
  'localized:boolean',
  'object',
  'array',
];

export const UIWidgetsByType = {
  string: ['upload', 'select'],
};

const localize = (text) =>
  text && typeof text === 'object' ? text.en || '' : text;
const localizeSchemaForm = (schema) => schema;

export function useLocalizedText(t, language) {
  return {
    localize,
    localizeSchemaForm,
  };
}
