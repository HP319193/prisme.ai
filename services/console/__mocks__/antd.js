export const notification = {
  success: jest.fn(),
  error: jest.fn(),
};

export function Spin({ children = null }) {
  return children;
}

export function Card({ children = null }) {
  return children;
}

export function Divider({ children = null }) {
  return children;
}

export function Switch({ children = null }) {
  return children;
}

export function Tooltip({ children = null }) {
  return children;
}

export function Select({ children = null }) {
  return children;
}
Select.Option = function Option({ children = null }) {
  return children;
};
Select.OptGroup = function OptGroup({ children = null }) {
  return children;
};

export function Panel({ children = null }) {
  return children;
}

export function Collapse({ children = null }) {
  return children;
}

export function Input({ children = null }) {
  return children;
}
Input.TextArea = function TextArea({ children = null }) {
  return children;
};

export function Typography({ children = null }) {
  return children;
}

export function Button({ children = null }) {
  return children;
}

export function DatePicker({ children = null }) {
  return children;
}

export function Popconfirm({ children = null }) {
  return children;
}

export function Menu({ children = null }) {
  return children;
}

export function Dropdown({ children = null }) {
  return children;
}
