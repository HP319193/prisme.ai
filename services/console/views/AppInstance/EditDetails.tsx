import {
  CloseCircleOutlined,
  DeleteOutlined,
  SettingOutlined,
} from '@ant-design/icons';
import {
  Button,
  Popover,
  Schema,
  SchemaForm,
  Tabs,
} from '@prisme.ai/design-system';
import { PopoverProps } from '@prisme.ai/design-system/lib/Components/Popover';
import { useTranslation } from 'next-i18next';
import { useCallback, useEffect, useMemo, useRef, useState } from 'react';
import { useTracking } from '../../components/Tracking';
import useLocalizedText from '../../utils/useLocalizedText';
import ConfirmButton from '../../components/ConfirmButton';
import { SLUG_VALIDATION_REGEXP } from '../../utils/regex';

interface EditDetailsprops extends Omit<PopoverProps, 'content'> {
  value: any;
  onSave: (values: any) => Promise<void | Record<string, string>>;
  onDelete: () => void;
  context?: string;
  disabled?: boolean;
}

export const EditDetails = ({
  value,
  onSave,
  onDelete,
  context,
  disabled,
  onOpenChange,
  ...props
}: EditDetailsprops) => {
  const { t } = useTranslation('workspaces');
  const { localize } = useLocalizedText();
  const { trackEvent } = useTracking();
  const [open, setOpen] = useState(false);
  const [values, setValues] = useState(value);

  const configSchema: Schema = useMemo(
    () => ({
      type: 'object',
      properties: {
        slug: {
          type: 'string',
          title: t('apps.details.slug.label'),
          pattern: SLUG_VALIDATION_REGEXP.source,
          errors: {
            pattern: t('automations.save.error_InvalidSlugError'),
          },
        },
        disabled: {
          type: 'boolean',
          title: t('apps.details.disabled.label'),
          description: t('apps.details.disabled.description'),
        },
      },
    }),
    [t]
  );

  const initialOpenState = useRef(false);
  useEffect(() => {
    if (!initialOpenState.current) {
      initialOpenState.current = true;
      return;
    }
    trackEvent({
      name: `${open ? 'Open' : 'Close'} Details Panel`,
      action: 'click',
    });
  }, [open, trackEvent]);

  const onChange = useCallback(
    (schema: Schema) => (changedValues: any) => {
      const newValues = { ...values };
      Object.keys(schema?.properties || {}).forEach((k) => {
        newValues[k] = changedValues[k];
      });
      setValues(newValues);
    },
    [values]
  );

  const submit = useCallback(() => {
    onSave(values);
  }, [onSave, values]);

  const buttons = useMemo(
    () => [
      <div key="1" className="flex flex-1 justify-end !mt-2 mx-4">
        <Button variant="primary" type="submit" disabled={disabled}>
          {t('save', { ns: 'common' })}
        </Button>
      </div>,
    ],
    [disabled, t]
  );

  return (
    <Popover
      titleClassName="flex m-0 pb-0 pt-4 pl-4 pr-4"
      title={({ setOpen }) => (
        <div className="flex flex-1 justify-between">
          {t('apps.details.title')}
          <button
            onClick={() => {
              trackEvent({
                name: 'Close Details Panel by clicking button',
                action: 'click',
              });
              setOpen(false);
            }}
          >
            <CloseCircleOutlined />
          </button>
        </div>
      )}
      destroyTooltipOnHide
      content={({ setOpen }) => (
        <Tabs
          className="flex flex-1"
          items={[
            {
              key: 'config',
              label: t('blocks.builder.setup.label'),
              children: (
                <SchemaForm
                  schema={configSchema}
                  initialValues={values}
                  onChange={onChange(configSchema)}
                  onSubmit={submit}
                  buttons={buttons}
                />
              ),
              active: true,
            },
          ]}
          tabBarExtraContent={
            <ConfirmButton
              onConfirm={onDelete}
              confirmLabel={t('apps.delete.confirm', {
                name: localize(value.name),
              })}
            >
              <DeleteOutlined className="translate-y-[-2px]" />
              <span className="flex">{t('apps.delete.label')}</span>
            </ConfirmButton>
          }
        />
      )}
      overlayClassName="min-w-[50%]"
      onOpenChange={(v) => {
        setOpen(v);
        onOpenChange?.(v);
      }}
      {...props}
    >
      <button type="button" className="text-lg text-gray focus:outline-none">
        <SettingOutlined />
      </button>
    </Popover>
  );
};

export default EditDetails;
