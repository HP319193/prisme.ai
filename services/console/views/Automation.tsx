import { useRouter } from "next/router";
import { ReactElement } from "react";
import getAutomationLayout from "../layouts/AutomationLayout";
import getLayout, { useWorkspace } from "../layouts/WorkspaceLayout";

export const Automation = () => {
  const { workspace } = useWorkspace();
  const {
    replace,
    query: { automationId },
  } = useRouter();
  replace(`/workspaces/${workspace.id}/automations/${automationId}/manifest`);

  return null;
};

Automation.getLayout = (page: ReactElement) =>
  getLayout(getAutomationLayout(page));

export default Automation;
