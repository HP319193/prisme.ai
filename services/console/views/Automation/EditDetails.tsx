import {
  CloseCircleOutlined,
  DeleteOutlined,
  SettingOutlined,
} from '@ant-design/icons';
import {
  Button,
  Popover,
  Schema,
  SchemaForm,
  Tabs,
} from '@prisme.ai/design-system';
import { PopoverProps } from '@prisme.ai/design-system/lib/Components/Popover';
import { useTranslation } from 'next-i18next';
import { useCallback, useEffect, useMemo, useRef, useState } from 'react';
import { useTracking } from '../../components/Tracking';
import useLocalizedText from '../../utils/useLocalizedText';
import ConfirmButton from '../../components/ConfirmButton';
import ArgumentsEditor from '../../components/SchemaFormBuilder/ArgumentsEditor';
import { SLUG_VALIDATION_REGEXP } from '../../utils/regex';

interface EditDetailsprops extends Omit<PopoverProps, 'content'> {
  value: any;
  onSave: (values: any) => Promise<void | Record<string, string>>;
  onDelete: () => void;
  context?: string;
  disabled?: boolean;
}

export const EditDetails = ({
  value,
  onSave,
  onDelete,
  context,
  disabled,
  onOpenChange,
  ...props
}: EditDetailsprops) => {
  const { t } = useTranslation('workspaces');
  const { localize } = useLocalizedText();
  const { trackEvent } = useTracking();
  const [open, setOpen] = useState(false);
  const [values, setValues] = useState(value);

  const configSchema: Schema = useMemo(
    () => ({
      type: 'object',
      properties: {
        name: {
          type: 'localized:string',
          title: t('automations.details.name.label'),
        },
        slug: {
          type: 'string',
          title: t('automations.details.slug.label'),
          pattern: SLUG_VALIDATION_REGEXP.source,
          errors: {
            pattern: t('automations.save.error_InvalidSlugError'),
          },
        },
        description: {
          type: 'localized:string',
          title: t('automations.details.description.label'),
          'ui:widget': 'textarea',
          'ui:options': { textarea: { rows: 6 } },
        },
      },
      'ui:options': {
        grid: [
          [['name', 'slug'], ['description']],
          [['arguments']],
          [['private']],
          [['disabled']],
        ],
      },
    }),
    [t]
  );

  const schemaSchema: Schema = useMemo(
    () => ({
      type: 'object',
      properties: {
        arguments: {
          'ui:widget': ArgumentsEditor,
        },
      },
    }),
    []
  );
  const advancedSchema: Schema = useMemo(
    () => ({
      type: 'object',
      properties: {
        private: {
          type: 'boolean',
          title: t('automations.details.private.label'),
          description: t('automations.details.private.description'),
        },
        disabled: {
          type: 'boolean',
          title: t('automations.details.disabled.label'),
          description: t('automations.details.disabled.description'),
        },
      },
      'ui:options': {
        grid: [
          [['name', 'slug'], ['description']],
          [['arguments']],
          [['private']],
          [['disabled']],
        ],
      },
    }),
    [t]
  );

  const initialOpenState = useRef(false);
  useEffect(() => {
    if (!initialOpenState.current) {
      initialOpenState.current = true;
      return;
    }
    trackEvent({
      name: `${open ? 'Open' : 'Close'} Details Panel`,
      action: 'click',
    });
  }, [open, trackEvent]);

  const onChange = useCallback(
    (schema: Schema) => (changedValues: any) => {
      const newValues = { ...values };
      Object.keys(schema?.properties || {}).forEach((k) => {
        newValues[k] = changedValues[k];
      });
      setValues(newValues);
    },
    [values]
  );

  const submit = useCallback(() => {
    onSave(values);
  }, [onSave, values]);

  const buttons = useMemo(
    () => [
      <div key="1" className="flex flex-1 justify-end !mt-2 mx-4">
        <Button variant="primary" type="submit" disabled={disabled}>
          {t('save', { ns: 'common' })}
        </Button>
      </div>,
    ],
    [disabled, t]
  );

  return (
    <Popover
      titleClassName="flex m-0 pb-0 pt-4 pl-4 pr-4"
      title={({ setOpen }) => (
        <div className="flex flex-1 justify-between">
          {t('automations.details.title')}
          <button
            onClick={() => {
              trackEvent({
                name: 'Close Details Panel by clicking button',
                action: 'click',
              });
              setOpen(false);
            }}
          >
            <CloseCircleOutlined />
          </button>
        </div>
      )}
      destroyTooltipOnHide
      content={({ setOpen }) => (
        <Tabs
          className="flex flex-1"
          items={[
            {
              key: 'config',
              label: t('automations.details.setup'),
              children: (
                <SchemaForm
                  schema={configSchema}
                  initialValues={values}
                  onChange={onChange(configSchema)}
                  onSubmit={submit}
                  buttons={buttons}
                />
              ),
              active: true,
            },
            {
              key: 'schema',
              label: t('automations.arguments.title'),
              children: (
                <SchemaForm
                  schema={schemaSchema}
                  initialValues={values}
                  onChange={onChange(schemaSchema)}
                  onSubmit={submit}
                  buttons={buttons}
                />
              ),
            },
            {
              key: 'advanced',
              label: t('automations.details.advanced'),
              children: (
                <SchemaForm
                  schema={advancedSchema}
                  initialValues={values}
                  onChange={onChange(advancedSchema)}
                  onSubmit={submit}
                  buttons={buttons}
                />
              ),
            },
          ]}
          tabBarExtraContent={
            <ConfirmButton
              onConfirm={onDelete}
              confirmLabel={t('automations.delete.confirm', {
                name: localize(value.name),
              })}
            >
              <DeleteOutlined />
              <span className="flex">{t('automations.delete.label')}</span>
            </ConfirmButton>
          }
        />
      )}
      overlayClassName="min-w-[50%]"
      onOpenChange={(v) => {
        setOpen(v);
        onOpenChange?.(v);
      }}
      {...props}
    >
      <button type="button" className="text-lg text-gray focus:outline-none">
        <SettingOutlined />
      </button>
    </Popover>
  );
};

export default EditDetails;
