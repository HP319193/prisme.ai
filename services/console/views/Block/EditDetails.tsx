import {
  CloseCircleOutlined,
  CodepenOutlined,
  DeleteOutlined,
  SettingOutlined,
} from '@ant-design/icons';
import {
  Button,
  FieldProps,
  Popover,
  Schema,
  SchemaForm,
  Tabs,
} from '@prisme.ai/design-system';
import { PopoverProps } from '@prisme.ai/design-system/lib/Components/Popover';
import { Trans, useTranslation } from 'next-i18next';
import { useCallback, useEffect, useMemo, useRef, useState } from 'react';
import { useTracking } from '../../components/Tracking';
import useLocalizedText from '../../utils/useLocalizedText';
import CSSEditor from '../Page/CSSEditor';
import ConfirmButton from '../../components/ConfirmButton';

interface EditDetailsprops extends Omit<PopoverProps, 'content'> {
  value: any;
  onSave: (values: any) => Promise<void | Record<string, string>>;
  onDelete: () => void;
  context?: string;
  disabled?: boolean;
}

export const EditDetails = ({
  value,
  onSave,
  onDelete,
  context,
  disabled,
  onOpenChange,
  ...props
}: EditDetailsprops) => {
  const { t } = useTranslation('workspaces');
  const { localize } = useLocalizedText();
  const { trackEvent } = useTracking();
  const [open, setOpen] = useState(false);
  const [values, setValues] = useState(value);

  const configSchema: Schema = useMemo(
    () => ({
      type: 'object',
      properties: {
        name: {
          type: 'localized:string',
          title: t('blocks.details.name.label'),
        },
        slug: {
          type: 'string',
          title: t('blocks.details.slug.label'),
        },
        description: {
          type: 'localized:string',
          title: t('blocks.details.description.label'),
          'ui:widget': 'textarea',
          'ui:options': { textarea: { rows: 10 } },
        },
      },
    }),
    [t]
  );
  const advancedSchema: Schema = useMemo(
    () => ({
      type: 'object',
      properties: {
        photo: {
          type: 'string',
          title: t('blocks.details.photo.label'),
          description: t('blocks.details.photo.description'),
          'ui:widget': 'upload',
          'ui:options': {
            upload: { accept: 'image/jpg,image/gif,image/png,image/svg' },
          },
        },
        url: {
          type: 'string',
          title: t('blocks.details.url.label'),
          description: (
            <Trans
              t={t}
              i18nKey="blocks.details.url.description"
              components={{
                a: <a target="_blank" />,
              }}
            />
          ),
          'ui:widget': 'upload',
          'ui:options': {
            upload: {
              accept: '.js',
              defaultPreview: (
                <CodepenOutlined className="text-4xl !text-gray-200 flex items-center" />
              ),
            },
          },
        },
      },
    }),
    [t]
  );
  const stylesSchema: Schema = useMemo(
    () => ({
      type: 'object',
      properties: {
        css: {
          type: 'string',
          'ui:widget': (props: FieldProps) => <CSSEditor {...props} opened />,
        },
      },
    }),
    []
  );

  const initialOpenState = useRef(false);
  useEffect(() => {
    if (!initialOpenState.current) {
      initialOpenState.current = true;
      return;
    }
    trackEvent({
      name: `${open ? 'Open' : 'Close'} Details Panel`,
      action: 'click',
    });
  }, [open, trackEvent]);

  const onChange = useCallback(
    (schema: Schema) => (changedValues: any) => {
      const newValues = { ...values };
      Object.keys(schema?.properties || {}).forEach((k) => {
        newValues[k] = changedValues[k];
      });
      setValues(newValues);
    },
    [values]
  );

  const submit = useCallback(() => {
    onSave(values);
  }, [onSave, values]);

  const buttons = useMemo(
    () => [
      <div key="1" className="flex flex-1 justify-end !mt-2 mx-4">
        <Button variant="primary" type="submit" disabled={disabled}>
          {t('details.save', { context: 'block' })}
        </Button>
      </div>,
    ],
    [disabled, t]
  );

  return (
    <Popover
      titleClassName="flex m-0 pb-0 pt-4 pl-4 pr-4"
      title={({ setOpen }) => (
        <div className="flex flex-1 justify-between">
          {t('details.title', { context })}
          <button
            onClick={() => {
              trackEvent({
                name: 'Close Details Panel by clicking button',
                action: 'click',
              });
              setOpen(false);
            }}
          >
            <CloseCircleOutlined />
          </button>
        </div>
      )}
      destroyTooltipOnHide
      content={({ setOpen }) => (
        <Tabs
          className="flex flex-1"
          items={[
            {
              key: 'config',
              label: t('blocks.builder.setup.label'),
              children: (
                <SchemaForm
                  schema={configSchema}
                  initialValues={values}
                  onChange={onChange(configSchema)}
                  onSubmit={submit}
                  buttons={buttons}
                />
              ),
              active: true,
            },
            {
              key: 'advanced',
              label: t('blocks.builder.advanced.label'),
              children: (
                <SchemaForm
                  schema={advancedSchema}
                  initialValues={values}
                  onChange={onChange(advancedSchema)}
                  onSubmit={submit}
                  buttons={buttons}
                />
              ),
            },
            {
              key: 'styles',
              label: t('blocks.builder.style.label'),
              children: (
                <SchemaForm
                  schema={stylesSchema}
                  initialValues={values}
                  onChange={onChange(stylesSchema)}
                  onSubmit={submit}
                  buttons={buttons}
                />
              ),
            },
          ]}
          tabBarExtraContent={
            <ConfirmButton
              onConfirm={onDelete}
              confirmLabel={t('blocks.builder.delete.confirm', {
                name: localize(value.name),
              })}
            >
              <DeleteOutlined />
              <span className="flex">{t('blocks.builder.delete.label')}</span>
            </ConfirmButton>
          }
        />
      )}
      overlayClassName="min-w-[50%]"
      onOpenChange={(v) => {
        setOpen(v);
        onOpenChange?.(v);
      }}
      {...props}
    >
      <button type="button" className="text-lg text-gray focus:outline-none">
        <SettingOutlined />
      </button>
    </Popover>
  );
};

export default EditDetails;
