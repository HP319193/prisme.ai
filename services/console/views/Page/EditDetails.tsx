import {
  CloseCircleOutlined,
  DeleteOutlined,
  SettingOutlined,
} from '@ant-design/icons';
import {
  Button,
  FieldProps,
  Popover,
  Schema,
  Tabs,
} from '@prisme.ai/design-system';
import { PopoverProps } from '@prisme.ai/design-system/lib/Components/Popover';
import { useTranslation } from 'next-i18next';
import { useCallback, useEffect, useMemo, useRef, useState } from 'react';
import { useTracking } from '../../components/Tracking';
import useLocalizedText from '../../utils/useLocalizedText';
import CSSEditor from './CSSEditor';
import ConfirmButton from '../../components/ConfirmButton';
import SchemaForm from '../../components/SchemaForm/SchemaForm';

interface EditDetailsprops extends Omit<PopoverProps, 'content'> {
  value: any;
  onSave: (values: any) => Promise<void | Record<string, string>>;
  onDelete: () => void;
  context?: string;
  disabled?: boolean;
}

export const EditDetails = ({
  value,
  onSave,
  onDelete,
  context,
  disabled,
  onOpenChange,
  ...props
}: EditDetailsprops) => {
  const { t } = useTranslation('workspaces');
  const { localize } = useLocalizedText();
  const { trackEvent } = useTracking();
  const [open, setOpen] = useState(false);
  const [values, setValues] = useState(value);

  const configSchema: Schema = useMemo(
    () => ({
      type: 'object',
      properties: {
        name: {
          type: 'localized:string',
          title: t('pages.details.name.label'),
        },
        slug: {
          type: 'string',
          title: t('pages.details.slug.label'),
        },
        description: {
          type: 'localized:string',
          title: t('pages.details.description.label'),
          'ui:widget': 'textarea',
          'ui:options': { textarea: { rows: 10 } },
        },
      },
    }),
    [t]
  );
  const lifecycleSchema: Schema = useMemo(
    () => ({
      type: 'object',
      properties: {
        onInit: {
          type: 'string',
          title: t('pages.blocks.settings.onInit.label'),
          description: t('pages.blocks.settings.onInit.description'),
          'ui:widget': 'autocomplete',
          'ui:options': {
            autocomplete: 'events:listen',
          },
        },
        automation: {
          type: 'string',
          title: t('pages.blocks.settings.automation.label'),
          description: t('pages.blocks.settings.automation.description'),
          'ui:widget': 'select',
          'ui:options': {
            from: 'automations',
            filter: 'endpoint',
          },
        },
        updateOn: {
          type: 'string',
          title: t('pages.blocks.settings.updateOn.label'),
          description: t('pages.blocks.settings.updateOn.description'),
          'ui:widget': 'autocomplete',
          'ui:options': {
            autocomplete: 'events:emit',
          },
        },
      },
    }),
    [t]
  );
  const stylesSchema: Schema = useMemo(
    () => ({
      type: 'object',
      properties: {
        styles: {
          type: 'string',
          'ui:widget': (props: FieldProps) => <CSSEditor {...props} opened />,
        },
      },
    }),
    []
  );

  const initialOpenState = useRef(false);
  useEffect(() => {
    if (!initialOpenState.current) {
      initialOpenState.current = true;
      return;
    }
    trackEvent({
      name: `${open ? 'Open' : 'Close'} Details Panel`,
      action: 'click',
    });
  }, [open, trackEvent]);

  const onChange = useCallback(
    (schema: Schema) => (changedValues: any) => {
      const newValues = { ...values };
      Object.keys(schema?.properties || {}).forEach((k) => {
        newValues[k] = changedValues[k];
      });
      setValues(newValues);
    },
    [values]
  );

  const submit = useCallback(() => {
    onSave(values);
  }, [onSave, values]);

  const buttons = useMemo(
    () => [
      <div key="1" className="flex flex-1 justify-end !mt-2 mx-4">
        <Button variant="primary" type="submit" disabled={disabled}>
          {t('save', { ns: 'common' })}
        </Button>
      </div>,
    ],
    [disabled, t]
  );

  return (
    <Popover
      titleClassName="flex m-0 pb-0 pt-4 pl-4 pr-4"
      title={({ setOpen }) => (
        <div className="flex flex-1 justify-between">
          {t('pages.details.title')}
          <button
            onClick={() => {
              trackEvent({
                name: 'Close Details Panel by clicking button',
                action: 'click',
              });
              setOpen(false);
            }}
          >
            <CloseCircleOutlined />
          </button>
        </div>
      )}
      destroyTooltipOnHide
      content={({ setOpen }) => (
        <Tabs
          className="flex flex-1"
          items={[
            {
              key: 'config',
              label: t('blocks.builder.setup.label'),
              children: (
                <SchemaForm
                  schema={configSchema}
                  initialValues={values}
                  onChange={onChange(configSchema)}
                  onSubmit={submit}
                  buttons={buttons}
                />
              ),
              active: true,
            },
            {
              key: 'lifecycle',
              label: t('blocks.builder.lifecycle.label'),
              children: (
                <SchemaForm
                  schema={lifecycleSchema}
                  initialValues={values}
                  onChange={onChange(lifecycleSchema)}
                  onSubmit={submit}
                  buttons={buttons}
                />
              ),
            },
            {
              key: 'styles',
              label: t('blocks.builder.style.label'),
              children: (
                <SchemaForm
                  schema={stylesSchema}
                  initialValues={values}
                  onChange={onChange(stylesSchema)}
                  onSubmit={submit}
                  buttons={buttons}
                />
              ),
            },
          ]}
          tabBarExtraContent={
            <ConfirmButton
              onConfirm={onDelete}
              confirmLabel={t('pages.delete.confirm', {
                name: localize(value.name),
              })}
            >
              <DeleteOutlined className="translate-y-[-2px]" />
              <span className="flex">{t('pages.delete.label')}</span>
            </ConfirmButton>
          }
        />
      )}
      overlayClassName="min-w-[50%]"
      onOpenChange={(v) => {
        setOpen(v);
        onOpenChange?.(v);
      }}
      {...props}
    >
      <button type="button" className="text-lg text-gray focus:outline-none">
        <SettingOutlined />
      </button>
    </Popover>
  );
};

export default EditDetails;
