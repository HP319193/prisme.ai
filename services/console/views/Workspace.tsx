import {
  Button,
  Col,
  MenuTab,
  Row,
  SidePanel,
  Tooltip,
} from '@prisme.ai/design-system';
import { useTranslation } from 'next-i18next';
import { useEffect, useMemo, useState } from 'react';
import { EventsViewer } from '../components/EventsViewer';
import getLayout from '../layouts/WorkspaceLayout';
import AutomationsSidebar from './AutomationsSidebar';
import IconApps from '../icons/icon-apps.svgr';
import IconAutomations from '../icons/icon-automations.svgr';
import IconPages from '../icons/icon-pages.svgr';
import AppsSidebar from './AppsSidebar';
import PagesSidebar from './PagesSidebar';
import { DoubleLeftOutlined, DoubleRightOutlined } from '@ant-design/icons';
import ShareWorkspace from '../components/Share/ShareWorkspace';
import Storage from '../utils/Storage';
import { useWorkspace } from '../components/WorkspaceProvider';

export const Workspace = () => {
  const { t } = useTranslation('workspaces');
  const { workspace, fullSidebar, setFullSidebar, setShare } = useWorkspace();
  const [sidebar, setSidebar] = useState(
    Storage.get('__workpaceSidebar') || 'automations'
  );

  useEffect(() => {
    Storage.set('__workpaceSidebar', sidebar);
  }, [sidebar]);

  useEffect(() => {
    setShare({
      label: t('workspace.share'),
      component: ShareWorkspace,
    });
  }, [setShare, t]);

  const menu = useMemo(
    () => [
      {
        label: (
          <div className="flex items-center">
            <div className="flex mr-2">
              <IconApps width={16} height={16} />
            </div>
            {t('apps.link')}
          </div>
        ),
        key: 'apps',
      },
      {
        label: (
          <div className="flex items-center">
            <div className="flex mr-2">
              <IconAutomations width={16} height={16} />
            </div>
            {t('automations.link')}
          </div>
        ),
        key: 'automations',
      },
      {
        label: (
          <div className="flex items-center">
            <div className="flex mr-2">
              <IconPages width={16} height={16} />
            </div>
            {t('pages.link')}
          </div>
        ),
        key: 'pages',
      },
    ],
    [t]
  );
  console.log('wp', workspace);

  if (!workspace) return null;

  return (
    <>
      <Row className="h-full">
        <Col span={14} className="flex h-full">
          <EventsViewer />
        </Col>
        <Col span={10} className="flex h-full">
          <SidePanel
            variant="squared"
            className={`
            absolute top-0 right-0 bottom-0 !bg-white
            ${fullSidebar ? 'w-[90vw] drop-shadow' : 'w-full'}
            transition-all ease-in duration-200`}
            Header={
              <div className="flex flex-row items-center h-[70px] justify-between border border-gray-200 border-solid !border-t-0">
                <MenuTab
                  items={menu}
                  selected={
                    (menu.find(({ key }) => sidebar === key) || menu[0]).key
                  }
                  onSelect={setSidebar}
                />
                <Tooltip
                  title={t(
                    fullSidebar
                      ? 'workspace.collapseSidebar'
                      : 'workspace.expandSidebar'
                  )}
                  placement="left"
                >
                  <Button
                    onClick={() => setFullSidebar(!fullSidebar)}
                    className="!text-sm"
                  >
                    {fullSidebar ? (
                      <DoubleRightOutlined
                        className="color-blue"
                        alt="workspace.collapseSidebar"
                      />
                    ) : (
                      <DoubleLeftOutlined
                        className="color-blue"
                        alt="workspace.expandSidebar"
                      />
                    )}
                  </Button>
                </Tooltip>
              </div>
            }
          >
            <>
              {sidebar === 'apps' && <AppsSidebar />}
              {sidebar === 'automations' && <AutomationsSidebar />}
              {sidebar === 'pages' && <PagesSidebar />}
            </>
          </SidePanel>
        </Col>
      </Row>
    </>
  );
};

Workspace.getLayout = getLayout;

export default Workspace;
