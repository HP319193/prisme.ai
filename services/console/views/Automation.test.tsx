import Automation from './Automation';
import renderer, { act } from 'react-test-renderer';
import AutomationBuilder from '../components/AutomationBuilder';
import { useRouter } from 'next/router';
import useKeyboardShortcut from '../components/useKeyboardShortcut';
import { notification } from '@prisme.ai/design-system';
import EditDetails from '../layouts/EditDetails';
import { useWorkspace } from '../providers/Workspace';
import { PageHeader } from 'antd';

jest.mock('../utils/useYaml', () => {
  const toJSON = jest.fn();
  const toYaml = jest.fn();
  const useYaml = jest.fn(() => ({
    toJSON,
    toYaml,
  }));
  return useYaml;
});

jest.mock('next/router', () => {
  const mock = {
    query: {
      automationId: 'foo',
    },
    replace: jest.fn(),
  };
  return {
    useRouter: () => mock,
  };
});

jest.mock('../components/useKeyboardShortcut', () => jest.fn());

jest.mock('../components/AutomationBuilder', () => () => null);

beforeEach(() => {
  useWorkspace().workspace = {
    id: '42',
    name: 'foo',
    createdAt: '',
    updatedAt: '',
    automations: {
      foo: {
        name: 'Hello',
        do: [],
      },
      bar: {
        name: 'World',
        do: [],
      },
    },
  };

  useRouter().query.automationId = 'foo';
});

it('should render', () => {
  const root = renderer.create(<Automation />);
  expect(root.toJSON()).toMatchSnapshot();
});

it('should render redirect', () => {
  useRouter().query.automationId = 'not found';
  const root = renderer.create(<Automation />);
  expect(root.toJSON()).toMatchSnapshot();
});

it('should update value', async () => {
  const root = renderer.create(<Automation />);
  expect(root.root.findByType(AutomationBuilder).props.value).toEqual({
    name: 'Hello',
    do: [],
  });

  (useRouter() as any).query.automationId = 'bar';

  act(() => {
    root.update(<Automation />);
  });

  expect(root.root.findByType(AutomationBuilder).props.value).toEqual({
    name: 'World',
    do: [],
  });
});

it('should change url after changing slug', async () => {
  const root = renderer.create(<Automation />);
  const title = renderer.create(root.root.findByType(PageHeader).props.title);

  act(() => {
    title.root
      .findByType(EditDetails)
      .props.onSave({ slug: 'foofoo', name: 'foofoo', description: '' });
  });

  expect(root.root.findByType(AutomationBuilder).props.value).toEqual({
    name: 'foofoo',
    slug: 'foofoo',
    description: '',
    do: [],
  });

  await act(async () => {
    await true;
  });

  expect(useRouter().replace).toHaveBeenCalledWith(
    '/workspaces/42/automations/foofoo',
    undefined,
    {
      shallow: true,
    }
  );
});

it('should save', async () => {
  const root = renderer.create(<Automation />);

  act(() => {
    return;
  });

  await act(async () => {
    await root.root.findByType(PageHeader).props.extra[0].props.onClick();
  });

  expect(useWorkspace().updateAutomation).toHaveBeenCalled();
  expect(notification.success).toHaveBeenCalledWith;
});

it('should save on shortcut', async () => {
  const root = renderer.create(<Automation />);

  act(() => {
    return;
  });
  const e = { preventDefault: jest.fn() };
  expect(useKeyboardShortcut).toHaveBeenCalled();
  await act(async () => {
    await (useKeyboardShortcut as jest.Mock).mock.calls[0][0][0].command(e);
  });
  expect(e.preventDefault).toHaveBeenCalled();
  expect(useWorkspace().updateAutomation).toHaveBeenCalled();
});
