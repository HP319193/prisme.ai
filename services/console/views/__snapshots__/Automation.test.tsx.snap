// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<div
  className="relative flex flex-1"
/>
`;

exports[`should render redirect 1`] = `
<div
  className="relative flex flex-1"
/>
`;
